test/e2e/e2e_test.go

```diff
@@ -162,65 +163,31 @@ func TestE2EFlow(t *testing.T) {
 		// also, verify that other user's resource are left intact
 		_, err = hostAwait.WaitForMasterUserRecord(extrajohnName)
 		require.NoError(t, err)
-		verifyResources(t, awaitility, extrajohnName, wait.UntilMasterUserRecordHasConditions(isProvisioned()))
+		verifyResources(t, awaitility, extrajohnName, wait.UntilMasterUserRecordHasConditions(provisioned()))
 	})
 }
 
-func setup(ctx *framework.TestCtx, awaitility *wait.Awaitility, username string) (*toolchainv1alpha1.UserSignup, *toolchainv1alpha1.UserAccountSpec, error) {
+func setup(ctx *framework.TestCtx, awaitility *wait.Awaitility, username string) (*toolchainv1alpha1.UserSignup, *toolchainv1alpha1.UserAccountSpec) {
 	// 0. Verify that the `basic` NSTemplateTier resource exists (will be needed later)
 	revisions, err := getRevisions(awaitility)
-	if err != nil {
-		return nil, nil, err
-	}
+	require.NoError(awaitility.T, err)
```

Comment:
I think we shouldn;t do that, because the `T` in `awaitility.T` refers to the parent `t` and in case of error, it may cause a panic

Reply:
awaitility.T refers to the same t actually. But even if it didn't (but it does) I thought that testify now properly handle the parent-child T situation.

Reply:
testify handles the parent-child change of T in https://github.com/stretchr/testify/pull/655/files#diff-da9b094bf431c6cb21dbc8467bc17c17R74 and you can see that `T` is set to the subtests' `t` in line 74. Is it the case for `awaitility`?

Reply:
Let's move the awaitility.T discussion to a separate PR. For now I just added t param to setup().

Reply:
https://github.com/codeready-toolchain/toolchain-e2e/pull/47/commits/4f1fbb01a91236fb4977a72435d1b0d7bd84864c

Reply:
cool, thanks for taking this request into account, @alexeykazakov